torch/csrc/autograd/python_function.cpp

```diff
@@ -865,12 +871,23 @@ std::pair<UnpackedInput, InputFlags> unpack_input(PyObject* args) {
       }
       Py_INCREF(Py_False);
       PyTuple_SET_ITEM(flags.needs_input_grad.get(), i, Py_False);
+
+      if (need_input) {
+        // The following conversion from PyObject to IValue is expensive
+        // Only do it if profiler is enabled and needs input shapes
+        auto match = torch::jit::tryToInferType(arg);
+        if (match.success()) {
+          unpacked.input_arguments.push_back(
+              torch::jit::toIValue(arg, match.type()));
```

Comment:
std::move?

Reply:
Since IValue here is a temporary variable, I do not think it needs std::move. I got a compile error if I add that:
moving a temporary object prevents copy elision